R/functions__lazy.R

```diff
@@ -81,43 +81,53 @@ pl_col = function(name = "", ...) {
 #' pl$lit(1:5)$cumulative_eval(pl$element()$first() - pl$element()$last()**2)$to_r()
 pl_element = function() pl$col("")
 
-#' pl$count
-#' @description Count the number of values in this column/context.
-#' @param column if dtype is:
-#' - Series: count length of Series
-#' - str: count values of this column
-#' - NULL: count the number of value in this context.
+#' Return the number of rows in the context.
 #'
+#' This is similar to `COUNT(*)` in SQL.
+#' @return [Expression][Expr_class] of data type [UInt32][pl_dtypes]
+#' @seealso
+#' - [`<Expr>$count()`][Expr_count]
+#' @examples
+#' df = pl$DataFrame(
+#'   a = c(1, 2, NA),
+#'   b = c(3, NA, NA),
+#'   c = c("foo", "bar", "foo")
+#' )
 #'
-#' @keywords Expr_new
+#' df$select(pl$len())
+pl_len = function() .pr$Expr$new_len()
+
+
+#' Return the number of non-null values in the column.
 #'
-#' @return Expr or value-count in case Series
+#' This function is syntactic sugar for `pl$col(...)$count()`.
 #'
-#' @examples
+#' Calling this function without any arguments returns the number of rows in the context.
+#' This way of using the function is deprecated.
+#' Please use [`pl$len()`][pl_len] instead.
 #'
+#' @inheritParams pl_head
+#' @inherit pl_len return
+#' @seealso
+#' - [`pl$len()`][pl_len]
+#' - [`<Expr>$count()`][Expr_count]
+#' @examples
 #' df = pl$DataFrame(
-#'   a = c(1, 8, 3),
-#'   b = c(4, 5, 2),
+#'   a = c(1, 2, NA),
+#'   b = c(3, NA, NA),
 #'   c = c("foo", "bar", "foo")
 #' )
-#' df$select(pl$len())
 #'
+#' df$select(pl$count("a"))
 #'
-#' df$group_by("c", maintain_order = TRUE)$agg(pl$len())
-pl_len = function(column = NULL) {
-  if (is.null(column)) {
-    return(.pr$Expr$new_len())
+#' df$select(pl$count(c("b", "c")))
+pl_count = function(...) {
+  if (missing(...)) {
+    warning("`pl$count()` is deprecated and will be removed in 0.15.0. Use `pl$len()` instead.")
```

Comment:
This condition can be removed now since the next release will probably be 0.15.0